cynic-codegen/src/enum_derive/input.rs

```diff
@@ -24,3 +27,23 @@ pub struct EnumDeriveVariant {
     #[darling(default)]
     pub(super) rename: Option<SpannedValue<String>>,
 }
+
+impl EnumDeriveInput {
+    pub fn graphql_type_name(&self) -> String {
+        // May be have a good way.
+        String::from(
+            &*(self
+                .graphql_type
+                .as_ref()
+                .map(|val| val.clone())
+                .unwrap_or(SpannedValue::from(self.ident.to_string()))),
+        )
+    }
+
+    pub fn graphql_type_span(&self) -> Span {
+        self.graphql_type
+            .as_ref()
+            .map(|val| val.span())
+            .unwrap_or(self.ident.span())
```

Comment:
Similar here, would use `unwrap_or_else`:
```suggestion
            .unwrap_or_else(|| self.ident.span())
```